<?xml version="1.0"?>
<launch>
    <arg name="max_pwm" default="20"/>
    <arg name="run_now" default="true"/>
    <arg name="path_file" default=""/>
    <!-- logging -->
    <arg name="is_publish_path" default="false"/>
    <arg name="is_log_path" default="false"/>
    <arg name="logs_directory" default="$(find jetson_car)/logs"/>    

    <!-- ZED-camera SLAM -->
    <include file="$(find zed_wrapper)/launch/zed.launch"/>
    <include file="$(find velodyne_pointcloud)/launch/VLP16_points.launch"/>
    <node pkg="tf" type="static_transform_publisher" name="zed_to_gps"  args="0 0 0 -1.57 0 0 zed_camera_center velodyne 10"/>
    <node name="map" pkg="jetson_car" type="mapper_node" output="screen">
        <param name="cell_size" value="0.1"/>
        <param name="distance_threshold" value="0.25"/>
        <param name="safe_radius" value="0.6"/>
        <param name="cnt_threshold" value="20"/>
    </node>
    <node name="planner" pkg="jetson_car" type="motion_planner_node" output="screen"/>

    <!-- publish path from file -->
    <include file="$(find jetson_car)/launch/include/_path_publisher.launch" if="$(eval path_file!='')">
        <arg name="path_file" value="$(arg path_file)"/>
    </include>

    <!-- logging -->
    <include file="$(find jetson_car)/launch/include/_logger.launch" if="$(eval is_publish_path or is_log_path)">
        <arg name="is_publish_path" value="$(arg is_publish_path)"/>
        <arg name="is_log_path" value="$(arg is_log_path)"/>    
        <arg name="logs_directory" value="$(arg logs_directory)"/>   
    </include>    

    <!-- motors control and message converter -->
    <node name="motors_driver" pkg="jetson_car" type="motors_driver_node" output="screen"/>    
    <node name="joy_to_motors" pkg="jetson_car" type="joy_to_motors_node.py" output="screen">
        <param name="max_pwm" value="$(arg max_pwm)"/>
    </node>

    <!-- The main node - move along the path -->
    <node name="path_mover" pkg="jetson_car" type="path_mover_node.py" if="$(arg run_now)" output="screen"/>    
</launch>
